### PR TITLE
Add reset cause to hwinfo for i.MX series

### PR DESCRIPTION
**Pull request by pdgendt**
This PR adds support for i.MX RT series status reset controller (SRC) to get the reset cause if `CONFIG_HWINFO` is enabled.

I can only test my soc (rt1064), if I need to enable others I would like to get a list of supported devices.

### CODE REVIEW

**Review by dleach02 — APPROVED** [review]

File: drivers/hwinfo/hwinfo_imxrt.c [hunk 1]
```diff
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT nxp_imx_src
```
Comment (henrikbrixandersen) [inline drivers/hwinfo/hwinfo_imxrt.c:7]: If the existing driver works _without_ requiring the SRC to be present, I think this functionality needs to go in as a different (dedicated) driver (unless all i.MX RT have the SRC IP, in which case I think it should be enabled on all those SoCs within this PR).

Device ID and reset cause does not need to be provided by a single HWINFO driver; see for instance the `CONFIG_HWINFO_MCUX_SIM` and `CONFIG_HWINFO_MCUX_RCM` drivers. Both the SIM and RCM are present on e.g. the NXP KE1xF, one providing a unique ID, the other a reset cause.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/hwinfo/hwinfo_imxrt.c (hunk 1 above)
Comment (pdgendt) [reply drivers/hwinfo/hwinfo_imxrt.c]: Thanks for the suggestion, @henrikbrixandersen 

I've moved the driver to a separate file, although I'm not entirely sure of the naming (`mcux/imx/rt`), maybe someone from NXP could suggest more appropriate file/config names if needed.

File: west.yml [hunk 2]
```diff
@@ -88,7 +88,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: c7bb88ec3240f1f44b1aafd2831b1472ca99b1d8
+      revision: pull/114/head
```
Comment (mmahadevan108) [inline west.yml:91]: This PR has been merged, please update this file

**Comment by mmahadevan108** [comment]
Any updates needed to the .yaml and documentation files in the `boards` folder?

**Comment by pdgendt** [comment]
> Any updates needed to the .yaml and documentation files in the `boards` folder?

I added `hwinfo` to the doc and yaml.

**Review by mmahadevan108 — APPROVED** [review]

**Comment by pdgendt** [comment]
I only updated the `rt1064` board, but many of the other i.MX boards probably can also enable this driver. As I don't have access to this information, I leave that to others to enable/test this.

**Comment by mmahadevan108** [comment]
> I only updated the `rt1064` board, but many of the other i.MX boards probably can also enable this driver. As I don't have access to this information, I leave that to others to enable/test this.

Could you please share how this driver was tested.

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by pdgendt** [comment]
> Could you please share how this driver was tested.

I simply call `hwinfo_get_reset_cause` and print out which bits are set during boot and verified these are as expected (pin reset/watchdog/..._)

**Comment by cfriedt** [comment]
@pdgendt - please rebase and resolve conflicts.

**Pull request merged**